src/Avalonia.Base/Platform/IMacOSTopLevelPlatformHandle.cs

```diff
@@ -8,6 +8,11 @@ public interface IMacOSTopLevelPlatformHandle
     {
         IntPtr NSView { get; }
         IntPtr GetNSViewRetained();
+    }
+
+    [Unstable]
+    public interface IMacOSWindowHandle : IMacOSTopLevelPlatformHandle
```

Comment:
I think it's a breaking change

Reply:
Can we mark methods as `[Obsolete(true)]` and return IntPtr.Zero for non-supporting implementations? Right now we'll be breaking completely unrelated code paths that are only concerned with actual windows.

Reply:
IMacOSTopLevelPlatformHandle and IMacOSWindowHandle shouldn't be Unstable, and instead should be NonClientImplementable, imo. It's not uncommon to use these APIs with interop.